src/fhirtypes/ElementDefinition.ts

```diff
@@ -802,8 +802,14 @@ export class ElementDefinition {
    * @returns {ElementDefinition[]} The elements at or inside of slices whose path matches the original element
    */
   findConnectedElements(postPath = ''): ElementDefinition[] {
-    const connectedElements = this.getSlices()
-      .filter(e => e.max !== '0') // we don't need zeroed-out slices
+    const slicedElement = this.slicedElement() ?? this;
+    let slicesToUse = slicedElement.getSlices();
+    if (this.sliceName) {
+      slicesToUse = slicesToUse.filter(slice => slice.sliceName.startsWith(`${this.sliceName}/`));
+    }
```

Comment:
> I feel like when this is a re-sliced element, calling this.getSlices() should get its own slices (the re-slices). In which case, the original logic would work. But you obviously introduced the new logic for a reason, so I must be missing something. If this.getSlices() does not return the re-slices when this is re-sliced, then is this the right place to correct that? Or should we be correcting getSlices()?

Previously `getSlices()` would always return an empty list when returned on a slice because it always assumed that `this.id` did not yet have a slice in it. So `getSlices()` would only get slices if `this` was a base element without a slice name. We tested it out, and it looks like it is safe to update how `getSlices()` is implemented so that it is always getting the slicier slices of `this`. So the behavior will remain the same for base elements, but `getSlices()` will now return any reslices of sliced elements.